test/Basic/Matrix/matrix_single_subscript_basic.test

```diff
@@ -0,0 +1,113 @@
+#--- source.hlsl
+RWBuffer<int> In : register(u0);
+RWBuffer<int> Index : register(u1);
+RWBuffer<int> Out : register(u2);
+RWBuffer<int> Out2 : register(u3);
+
+int4 getMatrix(int4x4 M, int index) {
+    return M[index];
+}
+
+[numthreads(1,1,1)]
+void main() {
+    int4 A = int4(In[0],  In[1],  In[2],  In[3]);
+    int4 B = int4(In[4],  In[5],  In[6],  In[7]);
+    int4 C = int4(In[8],  In[9],  In[10], In[11]);
+    int4 D = int4(In[12], In[13], In[14], In[15]);
+
+    int4x4 Mat;
+    Mat[Index[0]] = A;
+    Mat[Index[1]] = B;
+    Mat[Index[2]] = C;
+    Mat[Index[3]] = D;
+
+    int4x4 Mat2;
+    Mat2[Index[3]] = getMatrix(Mat, Index[0]);
```

Comment:
why the indirection here? why not inline the body of GetMatrix?

Reply:
sure